langlib/lang.annotations/src/main/ballerina/annotations.bal

```diff
@@ -65,7 +65,8 @@ const annotation ServiceIntrospectionDocConfig IntrospectionDocConfig on service
 # The usage of a deprecated program element is not recommended due to
 # various reasons. Hence, the compiler issues a warning when such an element is used.
 public const annotation deprecated on source type, source class, source const, source annotation,
-                source function, source parameter, source object function, source object field;
+                source function, source parameter, source object function, source object field,
+                source record field;
```

Comment:
Yes

Reply:
Is this from the spec?

Reply:
https://github.com/ballerina-platform/ballerina-spec/issues/823 This is an open spec issue for this. We have to wait until it resolved. Right?

Reply:
[annotation_access_ctx_config8.json](https://github.com/ballerina-platform/ballerina-lang/pull/37178/files#diff-2c76fbda2312dced76830eed796378b7928e3d8ea73f7e12d68de470047e4950) is failing due to this change(removing `object field` from `deprecated`). Is this the expected behaviour from FE? @prakanth97 @MaryamZi

Reply:
An annotation with `field` in its attach points applies to both record fields and object fields. This change basically allows what we allowed previously, but in addition also allows it on record fields. Wondering if we haven't considered this on the lang server side. 

Looping in @malinthar here, since he has raised the same in another comment.

Reply:
We can go ahead with `source object field, source record field` for now. Just noticed that there is a minor difference between field vs object field + record field which we may have to get clarified in the spec.

But generally, the behaviour is that they would apply to both kinds of fields.

Reply:
I see. I think we should add the FIELD attach point in the LS side by adding `annotationAttachPoints.add(AnnotationAttachPoint.FIELD);` after the following lines

https://github.com/ballerina-platform/ballerina-lang/blob/00388bb09dbf4f52cbf2da746ad9e805575c61ce/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/AnnotationAccessExpressionNodeContext.java#L134

https://github.com/ballerina-platform/ballerina-lang/blob/00388bb09dbf4f52cbf2da746ad9e805575c61ce/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/AnnotationAccessExpressionNodeContext.java#L142

Reply:
Updated a25087d0c83a9c1e4a1302185f42ffed9d0fd991

Reply:
@MaryamZi So with this change, `deprecated` annotation will be allowed on any `member-type-descriptor` as well (the difference you referred). It goes beyond https://github.com/ballerina-platform/ballerina-spec/issues/823 IMO.

Reply:
I think it's OK to allow it on tuple members, but we need to finalize this in the spec. Let's fall back to the original change for now.

Reply:
And yeah, field isn't record field + object field anymore since we use it with tuple members also.

Reply:
Agree

Reply:
@prakanth97 In `AnnotationAccessExpressionNodeContext.java` we should not add the attachPoint `RECORD_FIELD` (line 141-143) for the typeKind `RECORD` and same goes for line 134(remove the else block). Can you fix that?

Reply:
Updated eb5ea988cb6c1b1f8925377a0124368ee6f5377f